build.py

```diff
@@ -97,8 +100,8 @@ def check_c_lib(lib):
         with open(test_c_file, 'w') as fp:
             fp.write(textwrap.dedent(src))
 
-        return cmd_success('gcc %s -l%s -o %s' % \
-                (test_c_file, lib, test_e_file))
+        return cmd_success('gcc %s -l%s %s %s -o %s' % \
+                (test_c_file, lib, cxx_flags, ld_flags, test_e_file))
```

Comment:
If `cxx_flags` and `ld_flags` are lists, they should be converted to strings with `'  '.join()`?